src/commands/ip_allow.ts

```diff
@@ -0,0 +1,203 @@
+  writer(props).end(parse(data.project, props), {
+    fields: IP_ALLOW_FIELDS,
+  });
+};
+
+const add = async (
+  props: CommonProps &
+    ProjectScopeProps & {
+      ips: string[];
+      primaryOnly?: boolean;
+    },
+) => {
+  if (props.ips.length <= 0) {
+    log.error(`Enter individual IP addresses, define ranges with a dash, or use CIDR notation for more flexibility.
+       Example: neonctl ip-allow add 192.168.1.1, 192.168.1.20-192.168.1.50, 192.168.1.0/24 --project-id <id>`);
+    return;
+  }
+
+  const project: ProjectUpdateRequest['project'] = {};
+  const { data } = await props.apiClient.getProject(props.projectId);
+  const existingAllowedIps = data.project.settings?.allowed_ips;
+
+  project.settings = {
+    allowed_ips: {
+      ips: [...new Set(props.ips.concat(existingAllowedIps?.ips ?? []))],
+      primary_branch_only:
+        props.primaryOnly ?? existingAllowedIps?.primary_branch_only ?? false,
+    },
+  };
+
+  const { data: response } = await props.apiClient.updateProject(
+    props.projectId,
+    {
+      project,
+    },
+  );
+
+  writer(props).end(parse(response.project, props), {
+    fields: IP_ALLOW_FIELDS,
+  });
+};
+
+const remove = async (props: ProjectScopeProps & { ips: string[] }) => {
+  if (props.ips.length <= 0) {
+    log.error(
+      `Remove individual IP addresses and ranges. Example: neonctl ip-allow remove 192.168.1.1 --project-id <id>`,
+    );
+    return;
+  }
+
+  const project: ProjectUpdateRequest['project'] = {};
+  const { data } = await props.apiClient.getProject(props.projectId);
+  const existingAllowedIps = data.project.settings?.allowed_ips;
+
+  project.settings = {
+    allowed_ips: {
+      ips:
+        existingAllowedIps?.ips.filter((ip) => !props.ips.includes(ip)) ?? [],
+      primary_branch_only: existingAllowedIps?.primary_branch_only ?? false,
+    },
+  };
+
+  const { data: response } = await props.apiClient.updateProject(
+    props.projectId,
+    {
+      project,
+    },
+  );
+
+  writer(props).end(parse(response.project, props), {
+    fields: IP_ALLOW_FIELDS,
+  });
+};
+
+const reset = async (props: ProjectScopeProps & { ips: string[] }) => {
+  const project: ProjectUpdateRequest['project'] = {};
+  project.settings = {
+    allowed_ips: {
+      ips: props.ips,
+      primary_branch_only: false,
+    },
+  };
+
+  const { data } = await props.apiClient.updateProject(props.projectId, {
+    project,
+  });
+
+  writer(props).end(parse(data.project, props), {
+    fields: IP_ALLOW_FIELDS,
+  });
+
+  if (props.ips.length <= 0) {
+    log.info(
+      `The IP allowlist has been reset. All databases on project "${data.project.name}" are now exposed to the internet`,
+    );
+  }
+};
+
+const parse = (project: Project, props: CommonProps) => {
+  const ips = project.settings?.allowed_ips?.ips ?? [];
+  return {
+    id: project.id,
+    name: project.name,
+    IP_addresses: props.output === 'table' ? ips.join('\n') : ips,
```

Comment:
Let's add this to the writer? I'd debate that all array fields should be treated like this

Reply:
Moved to writer. Also added support for JSON Object in `table` output format.